### DISCUSSION TITLE
Integrating Current Density to Compute Current

### DISCUSSION THREAD

**Discussion by jsevic** [idaholab/moose#26034] [Q&A General] [answered]
Hello,

We have thin film structure, 2D at the moment, and have defined an AuxKernel that computes the vector current density. Our objective is to compute current over an edge boundary versus time.

I have been tried some of the post-processor integrators, such as FunctionSideIntegral and SideIntegration, and it's not clear to me I am getting what I expected. This was brought up in a previous thread #20280.

Do we need an Aux Kernel that computes current over time?

Any suggestions thx!

**Comment by lindsayad** [comment, marked as answer]
I would use a `SideIntegralVariablePostprocessor`

**Reply by jsevic** [reply]
Hello thanks I tried that and evidently it's looking for a scalar. Since my current density is computed as a vector do I need to form the dot product with the normal for the current through the boundary?

**Reply by lindsayad** [reply]
yes, sorry I missed that it was a vector. That makes sense!

**Reply by GiudGiud** [reply]
so this is effectively an advective flux
https://mooseframework.inl.gov/source/postprocessors/SideAdvectiveFluxIntegral.html

one thing though is that the postprocessing will only be correct if the discretization in the postprocessor and the kernels / BCs are the same

**Reply by lindsayad** [reply]
> will only be correct 

There's nuance here. For finite elements, as an example, if you use Dirichlet BCs then you no longer enforce global conservation and consequently a flux measured by a postprocessor might appear "wrong". However, it is still correct within finite element approximation error.

For a finite volume discretization, yes you should ensure that your forms in your postprocessor exactly match what's going on in the finite volume kernels/bcs

**Comment by jsevic** [comment]
Thanks using `SideIntegralVariablePostprocessor` works and I also used the AuxKernel `VectorVariableMagnitudeAux` to compute the magnitude of our current density, and the test cases produce what we expected.
